change(release): Document the release level for RPC and command-line changes (#7196)
ZcashFoundation/zebra: .github/PULL_REQUEST_TEMPLATE/release-checklist.md

Choose a release level for `zebrad`. Release levels are based on user-visible changes from the changelog:
- Mainnet Network Upgrades are `major` releases
- significant new features, large changes, deprecations, and removals are `minor` releases
    - any new, changed, or removed RPC methods or fields are a `minor` release

>     - any new, changed, or removed command-line arguments are a `minor` release

Author: @arya2 @gustavovalverde did you have any specific feedback or better wording about command-line changes?

Optional: Is there a way to summarise this better?

Reviewer: Should new config fields be a minor release too?

Author: > Should new config fields be a minor release too?

Yep, usually configs are a new feature, a new ability to configure an existing feature, or another behaviour change.

I think the IP connection limit in 1.0.1 was an exception, because the behaviour did change, but due to a security fix. And the config was intended for developers and testers only.

Author: I added "configs" to my suggestions in https://github.com/ZcashFoundation/zebra/pull/7196#discussion_r1258979877

> - significant new features, large changes, deprecations, and removals are `minor` releases
>     - any new, changed, or removed RPC methods or fields are a `minor` release
>     - any new, changed, or removed command-line arguments are a `minor` release

Reviewer: ```suggestion
- significant new features, changes to the RPC or command-line interfaces, large changes, deprecations, and removals are `minor` releases
```

Author: One-line suggestion:

```suggestion
- significant new features or behaviour changes; changes to RPCs, command-line, or configs; and deprecations or removals are `minor` releases
```

Author: Multi-line suggestion:

```suggestion
- `minor` releases are:
  - significant new features or behaviour changes,
  - changes to RPCs, command-line, or configs, or
  - deprecations or removals
```
